plugins/quotes.py

```diff
@@ -0,0 +1,217 @@
+			elif int(arg) < 0:
+				numarg = int(arg)
+			else:
+				numarg = 0
+		except:
+			wordarg = arg
+	
+	
+	quotelist = remove_quotes(quotelist)
+	
+	if len(quotelist) < 1:
+		asyncio.create_task(client.message(data.target, 'I have no quotes for '+target))
+		return
+		
+	if 'numarg' in locals():
+		try:
+			outquotes.append(quotelist[numarg])
+			if numarg < 0:
+				quotenums.append(len(quotelist)+numarg+1)
+			else:
+				quotenums.append(numarg+1)
+		except IndexError:
+			if len(quotelist) == 1:
+				asyncio.create_task(client.message(data.target, 'I only have 1 quote for '+target))
+			else:
+				asyncio.create_task(client.message(data.target, 'I only have '+str(len(quotelist))+' quotes for '+target))
+	elif 'wordarg' in locals():
+		for i in range(len(quotelist)):
+			quote = quotelist[i]
+			if wordarg in quote[3]:
+				outquotes.append(quote)
+				quotenums.append(i+1)
+	
+	"""if we have more than five quotes...."""
+	if len(outquotes) > 5:
+		"""append each quote with '\n', smack it right into hastebin"""
+		out = ''
+		for i in range(len(outquotes)):
+			print("posting...")
+			quote = outquotes[i]
+			out += '['+str(quotenums[i])+'/'+str(len(quotelist))+']'
+			out += ' <'+quote[1]+'> '+quote[3]+'\n'
+			
+		response = requests.post('https://hastebin.com/documents', data=out)
+		if response.status_code == 200:
+			d = response.json()['key']
+			link = 'https://hastebin.com/'+str(d)
+			asyncio.create_task(client.message(data.target, link))
+		else:
+			asyncio.create_task(client.message(data.target, 'Failed to upload quotes onto pastebin.'))
+	
+	else:
+		"""if not, send each quote as a seperate message"""
+		for i in range(len(outquotes)):
+			quote = outquotes[i]
+			out = '['+str(quotenums[i])+'/'+str(len(quotelist))+']'
+			out += ' <'+quote[1]+'> '+quote[3]
+			asyncio.create_task(client.message(data.target, out))
+	
+
+
+@hook.hook('command', ['qinit'], admin=True)
+async def quoteinit(client, data):
+	"""admin only quote db table initiation hook, since trying to init
+	on every quote(even though the db functions account for that) is kinda wasteful"""
+	conn = client.bot.dbs[data.server]
+	print ('Initializing quote table in /persist/db/'+data.server+'.db...')
+	db.init_table(conn, 'quotes', quote_columns)
+	db.ccache()
+	print ('Initialization complete.')
+
+@hook.hook('command', ['q'])
+async def quotes(client, data):
+	"""main quote hook""" 
+	conn = client.bot.dbs[data.server]
+	split = data.message.split()
+	
+	tables = db.get_table_names(conn)
+	if 'quotes' not in tables:
+		asyncio.create_task(client.message(data.target, 'Quote table uninitialized. Please ask your nearest bot admin to run .qinit.'))
+	
+	"""TODO: fix weird ass edgecase with users when a nick that corresponds to a command"""
+	
+	if len(split) > 1:
+		if split[0] == 'add':
+			quote = split.copy()
+			nick = quote[1]
+			
+			"""remove the 'add', and the nickname from the message"""
+			
+			quote.pop(0)
+			quote.pop(0)
+			
+			quote = ' '.join(quote)
```

Comment:
Why not just something like `' '.join(split[1:])` instead of making a copy and stuff? I'm sure indexing into split twice for the nick and quote is better than copying, assigning nick, then popping twice.